hardware_interface/include/hardware_interface/resource_manager.hpp

```diff
@@ -302,8 +302,13 @@ class HARDWARE_INTERFACE_PUBLIC ResourceManager
    */
   void write();
 
-  // Temporary method to keep old interface and reduce framework changes in PRs
-  void start_components();
+  /// Activates all available hardware components in the system.
+  /**
+   * All available hardware components int the ros2_control framework are activated.
+   * This is used to preserve default behavior from previous versions where all hardware components
+   * are activated per default.
+   */
+  void activate_all_components();
```

Comment:
very nice touch!

Reply:
Yeah, otherwise we have completely broken behavior. We need “spawners” for hardware interfaces soon. :D